src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -451,5 +451,69 @@ private void ProcessParameters(out bool omitInitialInfo)
                 throw new LoggerException(message, e, errorCode, helpKeyword);
             }
         }
+
+        private bool TryInterpretPathParameter(string parameter, out string filePath)
+        {
+            bool hasPathPrefix = parameter.StartsWith("LogFile=", StringComparison.OrdinalIgnoreCase);
+
+            bool isFileParam = hasPathPrefix || parameter.IndexOf('=') < 0;
+
+            if (!isFileParam)
+            {
+                filePath = null;
+                return false;
+            }
+
+            if (hasPathPrefix)
+            {
+                parameter = parameter.Substring("LogFile=".Length);
+            }
+
+            parameter = parameter.Trim('"');
+
+            bool isWildcard = ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_12) && parameter.Contains("{}");
+            bool hasProperExtension = parameter.EndsWith(".binlog", StringComparison.OrdinalIgnoreCase);
+            filePath = parameter;
+
+            if (!isWildcard)
+            {
+                return hasProperExtension;
+            }
+
+            filePath = parameter.Replace("{}", GetUniqueStamp(), StringComparison.Ordinal);
+
+            if (!hasProperExtension)
+            {
+                filePath += ".binlog";
+            }
+            return true;
+        }
+
+        private string GetUniqueStamp()
+            => PathParameterExpander.ExpandParameter(string.Empty);
+
+        private class BinlogPathParameterExpander : IBinlogPathParameterExpander
+        {
+            public string ExpandParameter(string parameters)
+                => $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}-{GenerateRandomString(6)}";
+
+            private string GenerateRandomString(int length)
```

Comment:
Shall we move this method to shared codebase helpers utils or something similar to have a possibility to reuse it by contributors? :)

Reply:
I moved it to Build.Framework